test/hummingbot/connector/exchange/kraken/test_kraken_exchange.py

```diff
@@ -0,0 +1,437 @@
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+        resp = self.get_open_orders_mock(quantity=1, price=2, order_type="buy")
+        mocked_api.post(regex_url, body=json.dumps(resp), callback=partial(self.register_sent_request, sent_messages))
+        userref = 1
+
+        ret = self.async_run_with_timeout(self.exchange.get_open_orders_with_userref(userref))
+
+        self.assertEqual(len(sent_messages), 1)
+
+        sent_message = sent_messages[0][1]["data"]
+
+        self.assertEqual(sent_message["userref"], userref)
+        self.assertEqual(ret, resp["result"])  # shallow comparison ok
+
+    @aioresponses()
+    def test_get_order(self, mocked_api):
+        sent_messages = []
+        order_id = "someId"
+        exchange_id = "someExchangeId"
+
+        url = f"{CONSTANTS.BASE_URL}{CONSTANTS.QUERY_ORDERS_PATH_URL}"
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+        resp = self.get_query_orders_mock(exchange_id, quantity=1, price=2, order_type="buy", status="closed")
+        mocked_api.post(regex_url, body=json.dumps(resp), callback=partial(self.register_sent_request, sent_messages))
+
+        self.exchange.start_tracking_order(
+            order_id=order_id,
+            exchange_order_id=exchange_id,
+            trading_pair=self.trading_pair,
+            trade_type=TradeType.BUY,
+            price=2,
+            amount=1,
+            order_type=OrderType.LIMIT,
+            userref=1,
+        )
+        ret = self.async_run_with_timeout(self.exchange.get_order(client_order_id=order_id))
+
+        self.assertEqual(len(sent_messages), 1)
+
+        sent_message = sent_messages[0][1]["data"]
+
+        self.assertEqual(sent_message["txid"], exchange_id)
+        self.assertEqual(ret, resp["result"])  # shallow comparison ok
+
+    @aioresponses()
+    def test_execute_buy(self, mocked_api):
+        self.exchange.start(self.clock, self.start_time)
+        self.simulate_trading_rules_initialized(mocked_api)
+
+        order_id = "someId"
+        exchange_id = "someExchangeId"
+        userref = 1
+        quantity = 1
+        price = 2
+
+        url = f"{CONSTANTS.BASE_URL}{CONSTANTS.ADD_ORDER_PATH_URL}"
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+        resp = self.get_order_placed_mock(exchange_id, quantity, price, order_type="buy")
+        mocked_api.post(regex_url, body=json.dumps(resp))
+
+        self.exchange.add_listener(MarketEvent.BuyOrderCreated, self.event_listener)
+        self.async_run_with_timeout(
+            self.exchange.execute_buy(
+                order_id,
+                self.trading_pair,
+                amount=Decimal(quantity),
+                order_type=OrderType.LIMIT,
+                price=Decimal(price),
+                userref=userref,
+            )
+        )
+
+        self.assertEqual(self.event_listener.events_count, 1)
```

Comment:
We should add assertions to check the event present in the listener is the expected one